app/src/gui/components/record/recordStepper.tsx

```diff
@@ -52,42 +55,196 @@ const useStyles = createUseStyles({
+ * @param {number} currentStep - The index of the currently active step.
+ * @param {string[]} stepperColors - Array of colors assigned to each step.
+ * @param {number[]} visitedSteps - List of steps that have been visited.
+ * @param {number[]} validSteps - List of steps marked as valid (no errors).
+ *
+ * @returns {string}
+ *   The color code for the step based on its state (error, valid, or default).
+ */
+const getStepColor = (
+  index: number,
+  totalSteps: number,
+  errors: any,
+  currentStep: number,
+  stepperColors: string[],
+  visitedSteps: number[],
+  validSteps: number[]
+) => {
+  const isVisited = visitedSteps.includes(index);
+  const isValid = validSteps.includes(index);
+  const hasErrors =
+    errors &&
+    Object.keys(errors).some(field => field.startsWith(`step_${index}`));
+
+  if (index === currentStep && hasErrors) return '#B10000';
+
+  if (hasErrors) return '#BDBDBD';
+
+  if (isVisited && isValid) return stepperColors[index];
+
+  if (isVisited && !isValid) return '#FFD700';
+
+  return '#BDBDBD';
+};
+
+/**
+ * @function RecordStepper
+ * @description
+ *   Displays a stepper component for multi-step forms, allowing navigation
+ *   between different views or sections. Handles visual indicators for visited steps,
+ *   validation status, and current active step. Adjusts layout responsively
+ *   for desktop and mobile views.
+ *
+ * @param {RecordStepperProps} props
+ *   - `view_index`: The index of the currently active view/step.
+ *   - `ui_specification`: The UI model containing view metadata and labels.
+ *   - `onChangeStepper`: Callback function to handle step changes.
+ *   - `views`: An array of view names representing form sections.
+ *   - `formErrors`: (Optional) Object representing validation errors for fields.
+ *
+ * @returns {JSX.Element}
+ *   A stepper UI for navigating through form sections with error indicators
+ *   and dynamic styling.
+ *
+ * @features
+ *   - Dynamic step color updates based on validation status.
+ *   - Tracks visited and valid steps.
+ *   - Responsive layout: horizontal stepper for large screens,
+ *     and mobile stepper for smaller screens.
+ */
+
 export default function RecordStepper(props: RecordStepperProps) {
   const classes = useStyles();
-  const {
-    view_index,
-    ui_specification,
-    onChangeStepper,
-    views, //add for branching logic
-  } = props;
-  // active step has been replaced by view_index because view_index will be updated every time form values updated
-  // 20220727 bbs the width 93% gets rid of the overflowX in the PSMIP notebook at most standard resolutions
-  // Client didn't want the absence of the stepper in sm-md resolutions, so reverted md->sm and am making text changes
+  const {view_index, ui_specification, onChangeStepper, views, formErrors} =
+    props;
+  const stepperColors = getStepperColors(views.length);
+
+  const [visitedSteps, setVisitedSteps] = useState<number[]>([]);
```

Comment:
You could definitely instantiate this for an existing form by at least including sections with responses as visited. But agree it's non-trivial.